main/lsp/lsp.h

```diff
@@ -63,18 +63,30 @@ class LSPLoop {
         CounterState counters;
     };
 
+    /**
+     * Encapsulates an update to LSP's file state.
+     */
+    struct FileUpdates {
+        std::vector<std::shared_ptr<core::File>> updatedFiles;
+        // Note: `string_view`s point to `string`s owned by `core::File`s in `updatedFiles`
+        std::vector<std::string_view> openedFiles;
```

Comment:
Note: I identify files with `string_view` because they might be new files, so they do not have a valid ID in `initialGS` yet.

Reply:
who owns the strings that those string_views come from? I think this is potential footgun. Could we make it store full `string`s instead?

Reply:
@DarkDimius see `// Note:` in source code.

Reply:
If it still makes you nervous, I can make them `string`s.

Reply:
can those vectors be vectors of `shared_ptr<File>` instead? So that they own the memory?

Reply:
(yeah, it does make me nervous)